netaddr.go

```diff
@@ -412,96 +323,82 @@ func (ip IP) Is6() bool {
 // returns the wrapped IPv4 address. Otherwise it returns ip, regardless
 // of its type.
 func (ip IP) Unmap() IP {
-	if !ip.Is4in6() {
-		return ip
+	if ip.Is4in6() {
+		ip.z = z4
 	}
-	a := ip.ipImpl.as16()
-	return IP{v4Addr{a[12], a[13], a[14], a[15]}}
+	return ip
 }
 
 // WithZone returns an IP that's the same as ip but with the provided
-// zone. If zone is empty, the zone is removed. If ip is an IPv4
+// zone. If zoneName is empty, the zone is removed. If ip is an IPv4
```

Comment:
Now that there's no `zone` type, this can go back to being `zone` instead of `zoneName`.

Reply:
Done.